04/SpeedTest.js

```diff
@@ -1,17 +1,86 @@
-import React, { useEffect } from 'react';
+import React, { useEffect, useState, useRef } from 'react';
 import useRandomItem from './hook';
 
 const SpeedTest = () => {
-    const [word, regenerateWord] = useRandomItem(['devmentor.pl', 'abc', 'JavaScript']);
+    const intervalRef = useRef(null);
+    const timeLimitSec = 10;
+    const [word, regenerateWord] = useRandomItem(['devmentor.pl', 'abc', 'JS', 'usb']);
+    const [text, setText] = useState('');
+    const [time, setTime] = useState(0);
+    const [points, setPoints] = useState(0);
+    const [length, setLength] = useState(0);
+    const [inFocus, setFocusState] = useState(false);
+
+    const stopInterval = () => clearInterval(intervalRef.current);
+    const startInterval = () => {
+        intervalRef.current = setInterval(() => {
+            setTime(step => {
+                if (step < timeLimitSec) {
+                    return step + 1;
+                }
+                stopInterval();
+                return 0;
+            }); // tutaj mnie linter grzecznie informuje o tym ze 'time' zostalo juz zadeklarowane w 'scope above', dlatego zmienilem na tic. jaka jest praktyka?
```

Comment:
Nie rozumiem. Nie widzę nigdzie `tic`?!

Reply:
poprawiłem to wcześniej a zapomniałem usunąć komentarza (: